src/hdf5_back.cc

```diff
@@ -532,6 +532,33 @@ QueryResult Hdf5Back::Query(std::string table, std::vector<Cond>* conds) {
               row[j] = x;
             break;
           }
+          case PAIR_INT_STRING: {
+            size_t nullpos;
+            unsigned int strlen = col_sizes_[table][j] - sizeof(int);
+            int xfirst = *reinterpret_cast<int*>(buf + offset);
```

Comment:
There is no avoiding it here. The C++ STL doesn't listen to the GCSG and the primary place this variable is used is in a string constructor (http://www.cplusplus.com/reference/string/string/string/) which requires an unsigned int. Good to know about this restriction and now I have another reason to hate the GCSG :).